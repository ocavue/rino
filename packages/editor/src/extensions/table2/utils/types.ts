import { FindProsemirrorNodeResult } from "@remirror/core"

export type FindTable = () => FindProsemirrorNodeResult | undefined
export interface CellAxis {
    row: number
    col: number
}
