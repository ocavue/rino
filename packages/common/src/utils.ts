export function isTestEnv() {
    return process.env.NODE_ENV === "development" || process.env.REACT_APP_TESTING
}
