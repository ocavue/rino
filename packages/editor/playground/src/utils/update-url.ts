interface URLOptions {
    contentId?: string
    enableDevTools?: boolean
}
export default function updateURLParams(o: URLOptions): void {
    const params = new URLSearchParams(document.location.search)
    if (o.contentId === undefined) {
        o.contentId = params.get("contentid") || "default"
    }
    if (o.enableDevTools === undefined) {
        o.enableDevTools = params.get("devtools") === "true"
    }
    const newParams = `?contentid=${o.contentId}&devtools=${o.enableDevTools ? "true" : "false"}`
    history.pushState(o, "", newParams)
}
