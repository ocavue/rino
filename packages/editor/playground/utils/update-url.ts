export function setURLParam(key: string, value: string): void {
    const params = new URLSearchParams(document.location.search)
    if (value) {
        params.set(key, value)
    } else {
        params.delete(key)
    }
    const search = params.toString()
    const url = document.location.pathname + (search ? "?" + search : "")
    history.pushState({}, "", url)
}

export function getURLParam(key: string, defaultValue = ""): string {
    const params = new URLSearchParams(document.location.search)
    return params.get(key) || defaultValue
}
