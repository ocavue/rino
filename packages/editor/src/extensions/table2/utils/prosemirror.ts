import { ProsemirrorNode } from "@remirror/pm"
import { Fragment, Slice } from "@remirror/pm/model"
import { Selection } from "@remirror/pm/state"
import { CellSelection } from "@remirror/pm/tables"
import { ReplaceAroundStep, Transform } from "@remirror/pm/transform"

type Attrs = Record<string, any>

// Change the attributes of the node at `pos`.
//
// This different between this function and `Transform.setNodeMarkup` is that it only overwrites fields that appear in `attrs`.
export function setNodeAttrs<T extends Transform>(tr: T, pos: number, attrs: Attrs, node?: ProsemirrorNode | null | undefined): T {
    node = node || tr.doc.nodeAt(pos)

    if (!node) {
        throw new RangeError("No node at given position")
    }

    const type = node.type
    const newNode = type.create({ ...node.attrs, ...attrs }, undefined, node.marks)

    if (node.isLeaf) {
        return tr.replaceWith(pos, pos + node.nodeSize, newNode)
    }

    if (!type.validContent(node.content)) {
        throw new RangeError(`Invalid content for node type ${type.name}`)
    }

    return tr.step(
        new ReplaceAroundStep(pos, pos + node.nodeSize, pos + 1, pos + node.nodeSize - 1, new Slice(Fragment.from(newNode), 0, 0), 1, true),
    )
}

// TODO: https://github.com/ProseMirror/prosemirror-tables/pull/126
export function selectionToCellSelection(selection: Selection): CellSelection {
    return selection as unknown as CellSelection
}

export function cellSelectionToSelection(selection: CellSelection): Selection {
    return selection as unknown as Selection
}
