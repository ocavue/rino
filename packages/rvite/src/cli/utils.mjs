import fs from "fs"
import { resolve } from "path"
const root = process.cwd()

const toAbsolute = (p) => resolve(root, p)

const filePaths = fs.readdirSync(toAbsolute("src/pages"))
// console.log("pagePaths:", filePaths)

function fileToUrl(filePath) {
    const name = filePath.replace(/\.(jsx|js|tsx|ts)$/, "").toLowerCase()
    return name === "index" ? `/` : `/${name}`
}

const routesToPrerender = filePaths.map(fileToUrl)

const fileToUrlMaps = {}
const urlToFileMaps = {}
const rollupInput = {}

for (const filePath of filePaths) {
    const url = fileToUrl(filePath)
    const file = `/src/pages/${filePath}`
    fileToUrlMaps[file] = url
    urlToFileMaps[url] = file

    let name = url.replace(/\/$/, "index").replace(/^\//, "")
    if (name === "_app") {
        continue
    }
    rollupInput[name] = "./" + name + ".html"
}

// console.log("routesToPrerender:", routesToPrerender)
// console.log("fileToUrlMaps:", fileToUrlMaps)
console.log("rollupInput:", rollupInput)

export { routesToPrerender, fileToUrlMaps, urlToFileMaps, rollupInput }
