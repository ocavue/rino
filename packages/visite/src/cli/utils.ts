import fs from "fs"
import { resolve } from "path"
const root = process.cwd()

const toAbsolute = (p: string) => resolve(root, p)

const filePaths = fs.readdirSync(toAbsolute("src/pages"))
// console.log("pagePaths:", filePaths)

function fileToUrl(filePath: string) {
    const name = filePath.replace(/\.(jsx|js|tsx|ts)$/, "").toLowerCase()
    return name === "index" ? `/` : `/${name}`
}

const routesToPrerender = filePaths.map(fileToUrl)

const fileToUrlMaps: Record<string, string> = {}
const urlToFileMaps: Record<string, string> = {}
const rollupInput: Record<string, string> = {}

for (const filePath of filePaths) {
    const url = fileToUrl(filePath)
    const file = `/src/pages/${filePath}`
    fileToUrlMaps[file] = url
    urlToFileMaps[url] = file

    const name = url.replace(/\/$/, "index").replace(/^\//, "")
    if (name === "_app") {
        continue
    }
    rollupInput[name] = "./" + name + ".html"
}

// console.log("routesToPrerender:", routesToPrerender)
// console.log("fileToUrlMaps:", fileToUrlMaps)
console.log("rollupInput:", rollupInput)

export { routesToPrerender, fileToUrlMaps, urlToFileMaps, rollupInput }
