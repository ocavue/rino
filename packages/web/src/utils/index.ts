export * from "./page"
export * from "./string"
export * from "./operation"
export * from "./env"
