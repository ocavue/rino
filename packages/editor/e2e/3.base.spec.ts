import "jest"

import { switchMode } from "./actions"
import { getSourceCodeModeText, setupEditor, type as typeByTestid } from "./utils"

beforeAll(async () => {
    await setupEditor()
})

async function type(text: string, pressEnter = true) {
    await typeByTestid("wysiwyg_mode_textarea", text, pressEnter)
}

describe("Source code text", () => {
    jest.setTimeout(180000)

    test("Create a note", async () => {
        await setupEditor("# ")
        await type("E2E test")
    })

    test("Inline text", async () => {
        await type("Text `Code` *Italic* **Strong** ***Italic+Strong***")
    })

    test("Headers", async () => {
        await type("# h1")
        await type("## h2")
        await type("### h3")
        await type("#### h4")
        await type("##### h5")
        await type("###### h6")
    })

    test("Ordered list", async () => {
        await type("1. First")
        await type("Second")
        await type("Third")
        await type("")
        await type("")
    })

    test("Quota", async () => {
        await type("Someone say:")
        await type("> Root 1")
        await type("Root 2")
        await type("> Sub 1")
        await type("Sub 2")
        await type("Sub 3")
        await type("")
        await type("Root 3")
        await type("Root 4")
        await type("")
        await type("And that's all")
    })

    test("Table", async () => {
        await type("| a | b | c |")
    })

    test("Check result", async () => {
        await switchMode() // Switch to the source code mode
        const received = await getSourceCodeModeText()
        expect(received.trim()).toMatchSnapshot()
        await switchMode() // Switch back to the WYSIWYG mode
    })
})
