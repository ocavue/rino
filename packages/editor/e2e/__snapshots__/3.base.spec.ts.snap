// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Source code text Check result 1`] = `
"# E2E test


Text \`Code\` *Italic* **Strong** ***Italic+Strong***


# h1


## h2


### h3


#### h4


##### h5


###### h6
1. First
2. Second
3. Third


Someone say:


> Root 1
>
>
> Root 2
>
>
> > Sub 1
> >
> >
> > Sub 2
> >
> >
> > Sub 3
>
>
> Root 3
>
>
> Root 4


And that's all


| a   | b   | c   |
| --- | --- | --- |
|     |     |     |"
`;
